client/src/modules/CourseManagement/components/ExpelledStudentsStats.tsx

```diff
@@ -0,0 +1,237 @@
+      key: 'date',
+      render: dateUtcRenderer,
+    },
+    {
+      title: 'Action',
+      key: 'action',
+      render: (_text, record) => (
+        <Button danger onClick={() => handleDelete(record.id)} loading={isDeleting}>
+          Del
+        </Button>
+      ),
+    },
+  ];
+
+  if (error) {
+    return <p>Failed to load statistics.</p>;
+  }
+
+  const [csvUrl, setCsvUrl] = React.useState<string | null>(null);
+  const downloadRef = React.useRef<HTMLAnchorElement>(null);
+
+  const handleExportCsv = () => {
+    if (!data || data.length === 0) {
+      return;
+    }
+
+    const exportableColumns = columns.filter(
+      (col): col is ColumnType<DetailedExpelledStat> => 'dataIndex' in col && col.dataIndex !== undefined,
+    );
+
+    const headers = exportableColumns
+      .map(col => {
+        if (Array.isArray(col.title)) {
+          return col.title.join(' ');
+        }
+        return col.title;
+      })
+      .filter(Boolean)
+      .join(',');
+
+    const csvRows = data.map(row => {
+      return exportableColumns
+        .map(col => {
+          let value = '';
+          const dataIndex = col.dataIndex;
+
+          if (Array.isArray(dataIndex)) {
+            let current: unknown = row;
+
+            for (const k of dataIndex) {
+              const keyAsString = String(k);
+
+              current = current ? (current as Record<string, unknown>)[keyAsString] : undefined;
+            }
+
+            value = current !== undefined && current !== null ? String(current) : '';
+          } else if (typeof dataIndex === 'string' || typeof dataIndex === 'number') {
+            value =
+              row[dataIndex as keyof DetailedExpelledStat] !== undefined &&
+              row[dataIndex as keyof DetailedExpelledStat] !== null
+                ? String(row[dataIndex as keyof DetailedExpelledStat])
+                : '';
+          } else if (col.key === 'reasons') {
+            value = row.reasonForLeaving ? row.reasonForLeaving.map(r => r.replace(/_/g, ' ')).join('; ') : '';
+          } else if (col.key === 'date') {
+            value = row.submittedAt ? new Date(row.submittedAt).toLocaleString() : '';
+          } else if (col.key === 'courseName') {
+            value = row.course ? row.course.alias : '';
+          } else if (col.key === 'githubId') {
+            value = row.user ? row.user.githubId : '';
+          }
+          if (value.includes(',') || value.includes('"')) {
+            return `"${value.replace(/"/g, '""')}"`;
+          }
+          return value;
+        })
+        .join(',');
+    });
+
+    const csvContent = [headers, ...csvRows].join('\n');
+    const blob = new Blob([csvContent], { type: 'text/csv;charset=utf-8;' });
+    const url = URL.createObjectURL(blob);
+    setCsvUrl(url);
+
+    setTimeout(() => {
+      if (downloadRef.current) {
+        downloadRef.current.click();
+        URL.revokeObjectURL(url);
+        setCsvUrl(null);
+      }
+    }, 0);
```

Comment:
I don't get why do we need this trick with zero `setTimeout` and `downloadRef` (and why do we need `downloadRef` as well here, it looks like some kind of not very reliable hack. Could you please clarify it and consider alternative approaches?